<a href="https://colab.research.google.com/github/mikeninerbravog/POC_ML_AI-in-Sustainability/blob/master/POC_ML_Sustainability.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
import random

# 1. Monitoramento Ambiental: Detecção de Desmatamento
def detectar_desmatamento():
    """
    Simula a detecção de desmatamento com base em dados de satélite.
    """
    np.random.seed(42)
    areas_monitoradas = [f"Área_{i}" for i in range(1, 21)]
    indice_vegetacao = np.random.uniform(0.2, 1.0, 20)  # Índice NDVI de vegetação (0 a 1)

    df = pd.DataFrame({"Área": areas_monitoradas, "Índice_Vegetação": indice_vegetacao})
    df["Alerta_Desmatamento"] = df["Índice_Vegetação"] < 0.5  # Ajustado para incluir True e False

    print("Áreas com possível desmatamento:\n", df[df["Alerta_Desmatamento"] == True])
    print("Áreas sem alerta de desmatamento:\n", df[df["Alerta_Desmatamento"] == False])
    return df

# 2. Otimização de Consumo de Energia
def prever_consumo_energia():
    """
    Simula um modelo de previsão de consumo energético para os próximos 10 dias.
    """
    np.random.seed(42)
    dias = np.arange(1, 101)
    consumo = 50 + np.cumsum(np.random.normal(0, 1, 100))  # Simula variação de consumo

    modelo = LinearRegression()
    modelo.fit(dias.reshape(-1, 1), consumo)

    # Previsão para os próximos 10 dias
    dias_futuros = np.arange(101, 111).reshape(-1, 1)
    previsao = modelo.predict(dias_futuros)

    print("Previsão de consumo de energia para os próximos 10 dias:", previsao)
    return previsao

# 3. Chatbot Sustentável Simples
def chatbot_sustentabilidade(pergunta):
    """
    Chatbot simples para responder dúvidas sobre sustentabilidade e IA.
    Esse chatbot pode ser expandido para um assistente interativo e autônomo.
    """
    respostas = {
        "Como a IA ajuda o meio ambiente?": "A IA monitora desmatamento, otimiza energia e reduz desperdícios.",
        "A IA pode prever desastres naturais?": "Sim, analisando dados climáticos e padrões de eventos anteriores.",
        "Como a IA reduz emissões de carbono?": "Ela ajuda indústrias a otimizar processos e minimizar desperdícios de energia.",
    }
    return respostas.get(pergunta, "Desculpe, não entendi sua pergunta.")

# Executando as funções
detectar_desmatamento()
prever_consumo_energia()

# Exemplo de interação com o chatbot
pergunta = "Como a IA ajuda o meio ambiente?"
print(f"Usuário: {pergunta}")
print(f"Chatbot: {chatbot_sustentabilidade(pergunta)}")


Áreas com possível desmatamento:
        Área  Índice_Vegetação  Alerta_Desmatamento
0    Área_1          0.499632                 True
4    Área_5          0.324815                 True
5    Área_6          0.324796                 True
6    Área_7          0.246467                 True
10  Área_11          0.216468                 True
13  Área_14          0.369871                 True
14  Área_15          0.345460                 True
15  Área_16          0.346724                 True
16  Área_17          0.443394                 True
19  Área_20          0.432983                 True
Áreas sem alerta de desmatamento:
        Área  Índice_Vegetação  Alerta_Desmatamento
1    Área_2          0.960571                False
2    Área_3          0.785595                False
3    Área_4          0.678927                False
7    Área_8          0.892941                False
8    Área_9          0.680892                False
9   Área_10          0.766458                False
11  Área_12  

### **Explicação dos Resultados**  

#### **1. Monitoramento de Desmatamento**
A função `detectar_desmatamento()` analisou **20 áreas** e classificou-as com base no **Índice de Vegetação (NDVI)**, que varia entre **0 e 1**:
- **Áreas com NDVI < 0.5** → **Marcadas como "Alerta de Desmatamento" (True)**.
- **Áreas com NDVI >= 0.5** → **Marcadas como "Sem Alerta de Desmatamento" (False)**.

##### **Áreas em alerta de desmatamento (`True`)**
| Área      | Índice de Vegetação (NDVI) | Alerta de Desmatamento |
|-----------|----------------------------|------------------------|
| Área_1    | **0.4996**                  | ✅ Sim (Alerta)        |
| Área_5    | **0.3248**                  | ✅ Sim (Alerta)        |
| Área_6    | **0.3247**                  | ✅ Sim (Alerta)        |
| Área_7    | **0.2464**                  | ✅ Sim (Alerta)        |
| Área_11   | **0.2164**                  | ✅ Sim (Alerta)        |
| Área_14   | **0.3698**                  | ✅ Sim (Alerta)        |
| Área_15   | **0.3454**                  | ✅ Sim (Alerta)        |
| Área_16   | **0.3467**                  | ✅ Sim (Alerta)        |
| Área_17   | **0.4433**                  | ✅ Sim (Alerta)        |
| Área_20   | **0.4329**                  | ✅ Sim (Alerta)        |

🔍 **Interpretação:**  
- O **NDVI baixo (<0.5)** indica possível **perda de vegetação**.
- Essas áreas precisam de **monitoramento contínuo** para avaliar riscos de desmatamento.

##### **Áreas sem alerta de desmatamento (`False`)**
| Área      | Índice de Vegetação (NDVI) | Alerta de Desmatamento |
|-----------|----------------------------|------------------------|
| Área_2    | **0.9605**                  | ❌ Não (Seguro)        |
| Área_3    | **0.7855**                  | ❌ Não (Seguro)        |
| Área_4    | **0.6789**                  | ❌ Não (Seguro)        |
| Área_8    | **0.8929**                  | ❌ Não (Seguro)        |
| Área_9    | **0.6808**                  | ❌ Não (Seguro)        |
| Área_10   | **0.7664**                  | ❌ Não (Seguro)        |
| Área_12   | **0.9759**                  | ❌ Não (Seguro)        |
| Área_13   | **0.8659**                  | ❌ Não (Seguro)        |
| Área_18   | **0.6198**                  | ❌ Não (Seguro)        |
| Área_19   | **0.5455**                  | ❌ Não (Seguro)        |

🔍 **Interpretação:**  
- O **NDVI alto (>0.5)** indica que **essas áreas estão com vegetação saudável**.
- Não há **necessidade imediata de intervenção ambiental**.

---

#### **2. Previsão de Consumo de Energia**
A função `prever_consumo_energia()` usou **Regressão Linear** para prever o **consumo energético nos próximos 10 dias**.

##### **Valores previstos para consumo de energia**
| Dia | Previsão de Consumo |
|-----|---------------------|
| 101 | **37.13**           |
| 102 | **36.99**           |
| 103 | **36.87**           |
| 104 | **36.74**           |
| 105 | **36.61**           |
| 106 | **36.48**           |
| 107 | **36.35**           |
| 108 | **36.23**           |
| 109 | **36.10**           |
| 110 | **35.97**           |

🔍 **Interpretação:**  
- O modelo prevê uma **pequena queda no consumo de energia** nos próximos dias.
- Essa previsão pode **ajudar na otimização do uso energético**, reduzindo desperdícios.

---

#### **3. Resposta do Chatbot sobre Sustentabilidade**
O chatbot foi consultado com a pergunta **"Como a IA ajuda o meio ambiente?"** e respondeu:
> *"A IA monitora desmatamento, otimiza energia e reduz desperdícios."*

Essa resposta está alinhada com as funcionalidades demonstradas:
1. **Monitoramento do meio ambiente** (detecção de desmatamento).
2. **Otimização do consumo de energia** (previsão de demanda energética).
3. **Redução de desperdícios** (tomada de decisão eficiente baseada em IA).

---

### **Conclusão**
✅ **A IA detectou com precisão as áreas com risco de desmatamento e as áreas preservadas.**  
✅ **O modelo previu o consumo de energia para os próximos 10 dias, mostrando uma leve tendência de queda.**  
✅ **O chatbot forneceu uma resposta coerente sobre IA e sustentabilidade.**

Esse modelo pode ser expandido para incluir **mais fatores ambientais**, como poluição do ar, qualidade da água e detecção de incêndios florestais.